### PR TITLE
Add end-to-end test to the daily NDMIS export job

### PR DESCRIPTION
**Pull request by sldblog**
## What does this pull request do?

🔎 **Please review commit-by-commit**

Adds an end-to-end test to the daily NDMIS export job.

To make the job output testable, it now [requires an `outputPath` parameter](https://github.com/ministryofjustice/hmpps-interventions-service/pull/1446/commits/eaa3e89dcccd07d8f00da54d41e667773ad4b75e).

Additionally, [remove warnings from BatchUtils](https://github.com/ministryofjustice/hmpps-interventions-service/commit/aaf18bfd79d4d1480514fa25d93e0186da156179).
- `LoggingWriter` was unused
- `.withRecordSeparator` was deprecated

## What is the intent behind these changes?

To make sure it works before it gets executed by a cronjob.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisPerformanceReportJobConfigurationTest.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import kotlin.io.path.fileSize
+import kotlin.io.path.listDirectoryEntries
+import kotlin.io.path.name
+import kotlin.io.path.pathString
+
+@Component
+class NdmisPerformanceJobLauncherTestUtils : JobLauncherTestUtils() {
+  @Autowired
+  override fun setJob(@Qualifier("ndmisPerformanceReportJob") job: Job) = super.setJob(job)
+}
```
Comment (sldblog) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisPerformanceReportJobConfigurationTest.kt:24]: This was frustrating. The only way I could get a specific job passed down to `JobLauncherTestUtils` if

1. Subclassed `JobLauncherTestUtils` (from the `org.springframework.batch:spring-batch-test` library)
2. Added a `@Qualifier` to `override fun setJob(...)`

File: build.gradle.kts [hunk 2]
```diff
@@ -104,5 +104,8 @@ dependencies {
   testImplementation("au.com.dius.pact.provider:junit5spring:4.4.4")
   testImplementation("com.squareup.okhttp3:okhttp:4.10.0")
   testImplementation("com.squareup.okhttp3:mockwebserver:4.10.0")
+  testImplementation("org.mockito:mockito-inline:5.0.0")
+  testImplementation("org.mockito:mockito-inline:5.0.0")
   testImplementation("org.mockito:mockito-inline:5.1.1")
```
Comment (sldblog) [inline build.gradle.kts:109]: wut

**Review by kth13 — APPROVED** [review]
LGTM

File: build.gradle.kts (hunk 2 above)
Comment (sldblog) [reply build.gradle.kts]: fixed + rebased

**Pull request merged**